code/modules/antagonists/heretic/magic/rust_construction.dm

```diff
@@ -43,26 +47,40 @@
 	invocation = span_danger("<b>[owner]</b> drags [owner.p_their()] hand[living_owner.usable_hands == 1 ? "":"s"] upwards as a wall of rust rises out of [cast_on]!")
 	invocation_self_message = span_notice("You drag [living_owner.usable_hands == 1 ? "a hand":"your hands"] upwards as a wall of rust rises out of [cast_on].")
 
-/datum/action/cooldown/spell/pointed/rust_construction/cast(turf/open/cast_on)
+/datum/action/cooldown/spell/pointed/rust_construction/cast(turf/cast_on)
 	. = ..()
 	var/rises_message = "rises out of [cast_on]"
+
+	// If we casted at a wall we'll try to rust it. In the case of an enchanted wall it'll deconstruct it
+	if(isclosedturf(cast_on))
+		cast_on.visible_message(span_warning("\The [cast_on] quakes as the rust causes it to crumble!"))
+		var/mob/living/living_owner = owner
+		living_owner?.do_rust_heretic_act(cast_on)
+		// ref transfers to floor
+		cast_on.Shake(shake_interval = 0.1 SECONDS, duration = 0.5 SECONDS)
+		// which we need to re-rust
+		living_owner?.do_rust_heretic_act(cast_on)
+		playsound(cast_on, 'sound/effects/bang.ogg', 50, vary = TRUE)
+		return
+
 	var/turf/closed/wall/new_wall = cast_on.place_on_top(/turf/closed/wall)
 	if(!istype(new_wall))
 		return
 
 	playsound(new_wall, 'sound/effects/constructform.ogg', 50, TRUE)
 	new_wall.rust_heretic_act()
 	new_wall.name = "\improper enchanted [new_wall.name]"
-	new_wall.hardness = 10
+	new_wall.AddComponent(/datum/component/torn_wall)
+	new_wall.hardness = 60
 	new_wall.sheet_amount = 0
 	new_wall.girder_type = null
 
 	// I wanted to do a cool animation of a wall raising from the ground
 	// but I guess a fading filter will have to do for now as walls have 0 depth (currently)
 	// damn though with 3/4ths walls this'll look sick just imagine it
 	new_wall.add_filter("rust_wall", 2, list("type" = "outline", "color" = "#85be299c", "size" = 2))
-	addtimer(CALLBACK(src, PROC_REF(fade_wall_filter), new_wall), filter_duration * (1/20))
-	addtimer(CALLBACK(src,PROC_REF(remove_wall_filter), new_wall), filter_duration)
+	addtimer(CALLBACK(src, PROC_REF(fade_wall_filter), new_wall), filter_duration * 0.5)
```

Comment:
Also `fade_wall_filter` has a value corresponding to the old value that you need to update `(19/20)`